protocol/handshaking/serverbound_handshake_test.go

```diff
@@ -1,6 +1,346 @@
 package handshaking
 
-import "testing"
+import (
+	"bytes"
+	"github.com/haveachin/infrared/protocol"
+	"net"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestServerBoundHandshake_Marshal(t *testing.T) {
+	tt := []struct {
+		packet          ServerBoundHandshake
+		marshaledPacket protocol.Packet
+	}{
+		{
+			packet: ServerBoundHandshake{
+				ProtocolVersion: 578,
+				ServerAddress:   "spook.space",
+				ServerPort:      25565,
+				NextState:       ServerBoundHandshakeStatusState,
+			},
+			marshaledPacket: protocol.Packet{
+				ID:   0x00,
+				Data: []byte{0xC2, 0x04, 0x0B, 0x73, 0x70, 0x6F, 0x6F, 0x6B, 0x2E, 0x73, 0x70, 0x61, 0x63, 0x65, 0x63, 0xDD, 0x01},
```

Comment:
Maybe a comment for what this data represents. This will be a nightmare for you or other maintainers in the future